Kitodo/src/main/java/org/kitodo/services/data/TaskService.java

```diff
@@ -700,6 +705,35 @@ private void abortTask(Task task) throws DataException {
         save(task);
     }
 
+    /**
+     * Performs creating images when this happens automatically in a task.
+     *
+     * @param executingThread
+     *            Executing thread (displayed in the taskmanager)
+     * @param task
+     *            Task that generates images
+     * @param automatic
+     *            Whether it is an automatic task
+     * @throws DataException
+     *             If the task cannot be saved
+     */
+    public void generateImages(EmptyTask executingThread, Task task, boolean automatic) throws DataException {
+        Process process = task.getProcess();
+        ImageGenerator generator = new ImageGenerator(new Subfolder(process, process.getProject().getGeneratorSource()),
+                GenerationMode.ALL, SubfolderFactoryService.createAll(process, task.getContentFolders()));
+        generator.setSupervisor(executingThread);
+        if (automatic) {
+            if (executingThread.getException() == null) {
+                task.setEditType(TaskEditType.AUTOMATIC.getValue());
+                task.setProcessingStatus(TaskStatus.DONE.getValue());
+            } else {
+                task.setEditType(TaskEditType.AUTOMATIC.getValue());
+                task.setProcessingStatus(TaskStatus.OPEN.getValue());
+                save(task);
```

Comment:
Why task is saved only in case of exception? If generation was successful there is no need to save task with processing status set up to done?

Reply:
Exciting question! I do not know either. But has always been like that and seems to work. Maybe it will then be saved elsewhere? Maybe it's also a bug. You can find out if you want it.

Reply:
You are re implementing it so you find out what is the correct way to do it.